#!/bin/bash
#SBATCH --nodes=1
#SBATCH --ntasks=11
#SBATCH -t 24:00:00

b=$SAMPLE

# Change enveomics & blastp paths to yours
enve=/project/mki314_uksr/enveomics/Scripts
blastp=/project/mki314_uksr/Software/ncbi-blast-2.15.0+/bin/blastp

# Source path to Conda environments
source /project/mki314_uksr/miniconda3/etc/profile.d/conda.sh

# These paths should remain the same
VFDB=$FOLDER/13.vfdb/VFDB_setA_pro_edited
ARGnorm=$FOLDER/metagenomic_analysis_pipeline_UKY/Readsbased_ARG_VF/zz.ARG_individual.py
ARGlength=$FOLDER/12.deep_arg/database/database/v2/features.gene.length
VFnorm=$FOLDER/metagenomic_analysis_pipeline_UKY/Readsbased_ARG_VF/zz.VFDB_individual.py
VFlength=$FOLDER/13.vfdb/VFDB_setA_pro_edited.gene.length

# The number of CPUs or threads
THR=10

#---------------------------------------------------------
# Run DeepARG
