#!/bin/bash
#SBATCH --nodes=1
#SBATCH --ntasks=4
#SBATCH -t 12:00:00

b=$SAMPLE
enve=/project/mki314_uksr/enveomics/Scripts
THR=4
BT=/project/mki314_uksr/miniconda3/envs/bowtie2/bin
ST=/project/mki314_uksr/miniconda3/envs/samtools/bin

#---------------------------------------------------------

# Run MetaBAT2 (Remove # to add metabat)

echo "==[ 15.metabat2: $(date) ]"
cd $FOLDER/15.metabat2

source /project/mki314_uksr/miniconda3/etc/profile.d/conda.sh
conda activate /project/mki314_uksr/miniconda3/envs/metabat2

# Run Bowtie2 alignment with all reads you have aginst one assembly

$BT/bowtie2-build ../05.assembly/"$b.LargeContigs.fna" "$b"

if [[ -s ../04.trimmed_fasta/"$b.SingleReads.fa" ]] ; then
   echo "Single Reads!"
   $BT/bowtie2 -f -x "$b" -U $FA -S "$b"_"$b"_Single.sam -p $THR;
 
else
   echo "Coupled Reads!"
   $BT/bowtie2 -f -x "$b" -1 ../04.trimmed_fasta/"$b"_1.fa -2 ../04.trimmed_fasta/"$b"_2.fa -S "$b"_"$b"_Coupled.sam -p $THR;
fi;

# Convert sam to bam and sort bam
$ST/samtools view -@ $THR -bS "$b"_"$b"_*.sam > "$b"_"$b".bam;
$ST/samtools sort -@ $THR "$b"_"$b".bam -o "$b"_"$b".bam.sorted;

#This is for multiple alignments for each contig (and make sure to change the alignment part above; use for loop there too).
#for i in ./"$b"_*.sam; do
#   m=$(basename $i .sam); 
#   $ST/samtools view -@ $THR -bS "$i" > "$m".bam;
#   $ST/samtools sort -@ $THR "$m".bam -o "$m".bam.sorted;
#done

# Create a depth file & Run metabat2

mkdir ./"$b"
jgi_summarize_bam_contig_depths --outputDepth "$b"_depth.txt "$b"_"$b".bam.sorted;

metabat2 -i ../05.assembly/"$b.LargeContigs.fna" -a "$b"_depth.txt -o ./"$b"/"$b"

# Remove all sam files

rm "$b"_*.sam

# remove all unsorted bam files

rm "$b"_*.bam

conda deactivate

#---------------------------------------------------------

cp $FOLDER/15.metabat2/$b/* $FOLDER/15.metabat2/binned/

#---------------------------------------------------------

## Prepare DasTool input

#conda activate DasTool

#cd $FOLDER/14.maxbin2

#Fasta_to_Scaffolds2Bin.sh -i ./"$b" -e fasta > ../16.dastool/"$b".maxbin.scaffolds2bin.tsv

#cd $FOLDER/15.metabat2

#Fasta_to_Scaffolds2Bin.sh -i ./"$b" -e fa > ../16.dastool/"$b".metabat.scaffolds2bin.tsv

## Run DasTool

#cd $FOLDER/16.dastool
#mkdir "$b"

#DAS_Tool --search_engine blast -i "$b".maxbin.scaffolds2bin.tsv,"$b".metabat.scaffolds2bin.tsv -l maxbin,metabat -c ../05.assembly/"$b".LargeContigs.fna -o ./"$b"

#---------------------------------------------------------

echo "Done: $(date)."
